diff --git a/ios/RNLaunchNavigator/LN_LaunchNavigator.m b/ios/RNLaunchNavigator/LN_LaunchNavigator.m
index 7b0072addf177349c5e54778721e0f98e3a00602..14630420473ba3d90e99cea26c06bb9ef7cdff1b 100644
--- a/ios/RNLaunchNavigator/LN_LaunchNavigator.m
+++ b/ios/RNLaunchNavigator/LN_LaunchNavigator.m
@@ -1320,13 +1320,17 @@ - (BOOL)isMapAppInstalled:(LNApp)mapApp {
     return [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:urlPrefix]];
 }
 
--(void) ensureNavigateParamNames{
-  for(id object in navigateParamNames){
-    NSString* key = object;
-    if(![navigateParams objectForKey:key]){
-      [navigateParams setValue:nil forKey:key];
+- (void)ensureNavigateParamNames {
+    NSMutableDictionary *mutableParams = [navigateParams mutableCopy];
+
+    for (NSString *key in navigateParamNames) {
+        if (![mutableParams objectForKey:key]) {
+            [mutableParams setObject:[NSNull null] forKey:key];
+        }
     }
-  }
+
+    // Reassign the modified copy back
+    navigateParams = [mutableParams copy];
 }
 
 - (NSString*)urlPrefixForMapApp:(LNApp)mapApp {
